src/d3a/models/strategy/area_agents/two_sided_pay_as_clear_engine.py

```diff
@@ -127,63 +127,115 @@ def _match_offers_bids(self):
             return
         clearing_rate, clearing_energy = clearing
         if clearing_energy > 0:
-            self.owner.log.info(f"Market Clearing Rate: {clearing_rate} "
-                                f"||| Clearing Energy: {clearing_energy} ")
+            self.owner.log.warn(f"Market Clearing Rate: {clearing_rate} "
+                                f"||| Clearing Energy: {clearing_energy} "
+                                f"||| Clearing Market {self.markets.source.area.name}")
             self.markets.source.state.clearing[time] = (clearing_rate, clearing_energy)
 
         cumulative_traded_bids = 0
+        accepted_bids = []
         for bid in self.sorted_bids:
-            already_tracked = self.owner.name == bid.buyer
+            original_bid_rate = bid.original_bid_price / bid.energy
             if cumulative_traded_bids >= clearing_energy:
                 break
-            elif (bid.price/bid.energy) >= clearing_rate and \
+            elif (bid.price / bid.energy) >= clearing_rate and \
                     (clearing_energy - cumulative_traded_bids) >= bid.energy:
                 cumulative_traded_bids += bid.energy
-                self.markets.source.accept_bid(
+                trade = self.markets.source.accept_bid(
                     bid=bid,
                     energy=bid.energy,
                     seller=self.owner.name,
```

Comment:
Sorry if I misunderstood. But isn't it that the bid's seller are still the clearing_engine owner's name. Meaning that bids are accepted in an aggregated fashion rather than being in P2P fashion

Reply:
The bids are never tracked (have a look at the already_tracked=True parameter). Therefore the bids are not being tracked, and the matching of the bids-offers take place exclusively on the offer side, by defining the offer.buyer according to the matched bid. Hope this makes it clearer.

Reply:
Can we discuss offline? May be I would better be able to explain?